doc/sphinx/src/fortran.rst

```diff
@@ -0,0 +1,693 @@
+.. _fortran:
+
+Fortran Interface
+================
+
+Singularity-EOS provides a Fortran interface which can be enabled with the CMake
+``SINGULARITY_USE_FORTRAN`` option. The interface provides Fortran bindings to the C++ EOS
+types and gives access to both initialization functions and EOS evaluation functions.
+
+The Fortran interface is defined in ``singularity-eos/eos/singularity_eos.f90``.
+
+
+Modules
+-------
+
+The Fortran interface consists of two modules:
+
+1. ``singularity_eos_types`` - Defines the data types used by the interface
+2. ``singularity_eos`` - Contains the functions for initializing and using EOS models
+
+Data Types
+----------
+
+.. code-block:: fortran
+
+   type :: sg_eos_ary_t
+     type(c_ptr) :: ptr = C_NULL_PTR
+   end type sg_eos_ary_t
+
+The ``sg_eos_ary_t`` type is a wrapper around a C pointer to the underlying EOS object.
+
+Initialization Functions
+-----------------------
+
+The following functions are available for initializing EOS models:
+
+.. code-block:: fortran
+
+   integer function init_sg_eos_f(nmat, eos) result(err)
+     integer(kind=c_int), intent(in) :: nmat
+     type(sg_eos_ary_t), intent(in)  :: eos
+   end function init_sg_eos_f
+
+Initializes the EOS array with the specified number of materials.
```

Comment:
There should probably be some discussion that only the EOS's available in the variant are defined, which depends on how the code is built. For example `init_sg_SpinerDependsRhoT` is only available if the code is built with spiner and HDF5, EOSPAC is only built if EOSPAC is enabled, and similarly Helmholtz is only built if Helmholtz is enabled.